### PR TITLE
Upgraded to babel6 and allowed for server-side rendering

### PR DESCRIPTION
**Pull request by sjeanat**
@jrowny @okonet  Hi! I wanted to use your draggable grid, but ran into some syntax issues, babel 6 issues, and server-side rendering issues. So, I upgraded to babel-6, made a few syntax updates, and made some further modifications to allow server-side rendering (i.e. change life-cycle method `componentDidMount` to `componentWillUpdate(nextProps)`) 

Please let me know if we can merge this or if you have any further suggestions. Thank you!

### CODE REVIEW

File: lib/BaseDisplayObject.jsx [hunk 1]
```diff
@@ -43,11 +48,15 @@ export default class BaseDisplayObject extends React.Component {
     return this.props.style;
   }
 
-  componentDidMount() {
-    if (this.props.dragEnabled) {
+  componentWillUpdate(nextProps) {
```
Comment (jrowny) [inline lib/BaseDisplayObject.jsx:51]: this could cause duplicate listeners to be added

File: lib/AbsoluteGrid.jsx [hunk 2]
```diff
@@ -10,16 +10,18 @@ export default class AbsoluteGrid extends React.Component {
 
   constructor(props, context){
     super(props, context);
+    this.getDOMWidth = this.getDOMWidth.bind(this);
+    this.onResize = this.onResize.bind(this);
     this.onResize = debounce(this.onResize, 150);
     this.dragManager = new DragManager(this.props.onMove, this.props.keyProp);
     this.state = {
-      layoutWidth: 0,
+      layoutWidth: 1400,
```
Comment (jrowny) [inline lib/AbsoluteGrid.jsx:18]: curious how you came up with this number

File: lib/AbsoluteGrid.jsx [hunk 3]
```diff
@@ -106,15 +108,15 @@ export default class AbsoluteGrid extends React.Component {
     window.removeEventListener('resize', this.onResize);
   }
 
-  onResize = () => {
+  onResize() {
```
Comment (jrowny) [inline lib/AbsoluteGrid.jsx:111]: I much prefer the fat arrow binding style to re-assigning `this.onResize.bind` in the constructor. I think this is a step backwards.

**Comment by jrowny** [comment]
thanks for contributing. The changes you made should not be needed for Babel 6, not sure why you're having issues, I'll try to investigate but I use babel 6 on other projects with the fat arrow binding class functions.

**Comment by sjeanat** [comment]
@jrowny ok, I guess I can't add server side rendering to this then. It's alright though, I have an alternative drag drop that will work too. But thank you so much anyways for taking the time.

**Comment by jrowny** [comment]
You certainly can add server side rendering, you just have to handle to check if the dragEnabled prop actually changed before adding listeners. That's the only issue. You're only checking if it is true, which means every-time the props change, if it's true, it adds another listener.

**Pull request closed without merging**